contracts/account/utils/draft-ERC7579Utils.sol

```diff
@@ -169,12 +172,38 @@ library ERC7579Utils {
     }
 
     /// @dev Decodes a batch of executions. See {encodeBatch}.
+    ///
+    /// NOTE: This function runs some checks and will throw a {ERC7579DecodingError} if the input is not properly formatted.
     function decodeBatch(bytes calldata executionCalldata) internal pure returns (Execution[] calldata executionBatch) {
-        assembly ("memory-safe") {
-            let ptr := add(executionCalldata.offset, calldataload(executionCalldata.offset))
-            // Extract the ERC7579 Executions
-            executionBatch.offset := add(ptr, 32)
-            executionBatch.length := calldataload(ptr)
+        unchecked {
+            uint256 bufferLength = executionCalldata.length;
+
+            // Check executionCalldata is not empty.
+            if (bufferLength < 32) revert ERC7579DecodingError();
+
+            // Get the offset of the array.
+            uint256 offset = uint256(bytes32(executionCalldata[0:32]));
```

Comment:
Seppi proposed renaming that to avoid confusion. Any idea for a better name @ernestognw @frangio ? he proposed `offsetToArrayLength`, I think its quite long but if you think its the better option I won't oppose it.

Reply:
I was thinking of `arrayOffset` but just read Seppi's feedback:

> but then arrayOffset points to the length, while array.offset points to the data, which I find confusing.

Let's use `arrayLengthOffset`